quickwit-metastore/src/metastore/single_file_metastore.rs

```diff
@@ -71,6 +72,7 @@ impl SingleFileMetastore {
         SingleFileMetastore {
             storage,
             cache: Arc::new(RwLock::new(HashMap::new())),
```

Comment:
Do we still need that lock if all the ops are behind a mutex?

Reply:
`cache` is wrapped in `Arc` to safely share ownership between threads.
However, `cache` wrapped with `Arc` can only be accessed immutable. To be able to change the value from multiple threads, I think need to use `RwLock` or `Mutex`.
Do you know a better way?

Reply:
@guilload but the borrow checker will never accept that right?

So  should we use the rwlock as a the mutex?

Reply:
Right, @fulmicoton. What I was trying to say is that we can do with one lock and a simple design to do so is to encapsulate the mutable state in some `inner` struct enclosed in a `Mutex`.

Reply:
@fulmicoton @guilload Thank you for the suggestion.